devito/equation.py

```diff
@@ -50,15 +57,33 @@ class Eq(sympy.Eq):
     def __new__(cls, *args, **kwargs):
         kwargs['evaluate'] = False
         subdomain = kwargs.pop('subdomain', None)
+        coefficients = kwargs.pop('coefficients', None)
         obj = sympy.Eq.__new__(cls, *args, **kwargs)
         obj._subdomain = subdomain
+        obj._coefficients = coefficients
+        functions = retrieve_functions(obj)
+        functions = filter_ordered(functions, key=lambda i: i.name)
+        functions = [f for f in functions if not f.is_SparseFunction]
+        if any(f.coefficients == 'symbolic' for f in functions):
+            # NOTE: As Coefficients.py is expanded we will not want
+            # all rules to be expunged during this procress.
+            rules = default_rules(obj, functions)
+            try:
+                obj = obj.xreplace({**coefficients.rules, **rules})
```

Comment:
what could possibly go wrong here? the ``except`` below needs to be specialised for code self-documentation . 
Also, I'm not sure I follow the logic underpinning this try-except: is it the idea that ``rules`` will eventually have to be applied?

Reply:
If `coefficients` is `None` (i.e. the user has not supplied a `CoefficentRules` object to this particular equation) then `coefficient.rules` will result in an attribute error. 

> Also, I'm not sure I follow the logic underpinning this try-except: is it the idea that rules will 
> eventually have to be applied?

In essence yes, it's saying try and replace coefficients with supplied rules + any default rules necessary, and if no rules have been supplied replace everything with default rules.

Apologies, what do you mean by 'specialised'? I'll look through other existing try/except instances to try and work that out shortly anyway.

Reply:
I see what you mean, except updated.